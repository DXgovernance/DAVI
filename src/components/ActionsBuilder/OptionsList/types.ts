import { Option } from '../types';
export interface OptionsListProps {
  isEditable: boolean;
  options: Option[];
  onChange: (options: Option[]) => void;
  addOption: () => void;
  editOption: (option: Option) => void;
}

export enum SimulationState {
  none,
  pending,
  allPassed,
  someFailed,
  error,
}
