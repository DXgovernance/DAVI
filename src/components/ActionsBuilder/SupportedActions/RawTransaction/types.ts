import { BigNumber } from 'ethers';

export interface RawTransactionValues {
  to: string;
  value: BigNumber;
  data: string;
}
