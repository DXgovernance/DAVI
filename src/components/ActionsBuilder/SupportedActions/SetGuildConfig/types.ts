import { BigNumber } from 'ethers';

export interface SetGuildConfigFields {
  proposalTime: BigNumber;
  timeForExecution: BigNumber;
  votingPowerPercentageForProposalExecution: BigNumber;
  votingPowerPercentageForProposalCreation: BigNumber;
  voteGas: BigNumber;
  maxGasPrice: BigNumber;
  maxActiveProposals: BigNumber;
  lockTime: BigNumber;
  minimumMembersForProposalCreation: BigNumber;
  minimumTokensLockedForProposalCreation: BigNumber;
}

export enum FieldType {
  duration = 'duration',
  number = 'number',
  percentage = 'percentage',
}
export interface Field {
  name: string;
  label: string;
  type: FieldType;
}

export type ControlField =
  | 'votingPowerPercentageForProposalExecution'
  | 'votingPowerPercentageForProposalCreation'
  | 'voteGas'
  | 'maxGasPrice'
  | 'maxActiveProposals'
  | 'minimumMembersForProposalCreation'
  | 'minimumTokensLockedForProposalCreation'
  | 'proposalTime'
  | 'timeForExecution'
  | 'lockTime';
