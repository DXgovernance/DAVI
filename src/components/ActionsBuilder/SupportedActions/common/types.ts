import { DecodedCall } from 'components/ActionsBuilder/types';

export interface SummaryProps {
  decodedCall: DecodedCall;
}
