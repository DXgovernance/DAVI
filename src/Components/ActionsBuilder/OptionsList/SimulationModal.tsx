import { Flex } from 'Components/Primitives/Layout';
import { Modal } from 'Components/Primitives/Modal';
import PendingCircle from 'old-components/Guilds/common/PendingCircle';
import { FiX, FiCheck } from 'react-icons/fi';
import {
  Message,
  Muted,
  StatusWrapper,
  SuccessCircle,
  TenderlyLogo,
  WarningCircle,
} from './SimulationModal.styled';
import { Button } from 'old-components/Guilds/common/Button';
import { useTranslation } from 'react-i18next';
import tenderlyLogo from 'assets/images/tenderly-dark-mode.svg';
import { SimulationState } from './OptionsList';

interface SimulationModalProps {
  isOpen: boolean;
  onDismiss: () => void;
  status: SimulationState;
}

const SimulationModal: React.FC<SimulationModalProps> = ({
  isOpen,
  onDismiss,
  status,
}) => {
  const { t } = useTranslation();

  return (
    <Modal
      isOpen={isOpen}
      onDismiss={onDismiss}
      header={t('simulations.verifyingActions')}
      maxWidth={390}
      zIndex={1000}
    >
      <Flex padding={`1.5rem`}>
        <StatusWrapper>
          <Flex>
            {status === SimulationState.pending ? (
              <SimulationPending />
            ) : status === SimulationState.allPassed ? (
              <SimulationPassed />
            ) : status === SimulationState.someFailed ? (
              <SimulationFailed />
            ) : (
              <ChainNotSupported />
            )}
          </Flex>
          <Flex direction="row">
            <Muted>{t('poweredBy')}</Muted> <TenderlyLogo src={tenderlyLogo} />
          </Flex>
        </StatusWrapper>
        <Button disabled={false} fullWidth onClick={onDismiss}>
          {t('dismiss')}
        </Button>
      </Flex>
    </Modal>
  );
};

export default SimulationModal;

const SimulationPending = () => {
  const { t } = useTranslation();

  return (
    <>
      <PendingCircle height="86px" width="86px" />
      <Message>{t('simulations.verifyingActions')}</Message>
    </>
  );
};

const SimulationPassed = () => {
  const { t } = useTranslation();
  return (
    <>
      <SuccessCircle>
        <FiCheck size={40} />
      </SuccessCircle>
      <Message>{t('simulations.actionsVerified')}</Message>
    </>
  );
};

const SimulationFailed = () => {
  const { t } = useTranslation();
  return (
    <>
      <WarningCircle>
        <FiX size={40} />
      </WarningCircle>
      <Message>{t('simulations.actionsFailed')}</Message>
    </>
  );
};

const ChainNotSupported = () => {
  const { t } = useTranslation();
  return (
    <>
      <WarningCircle>
        <FiX size={40} />
      </WarningCircle>
      <Message>{t('simulations.chainNotSupported')}</Message>
    </>
  );
};
