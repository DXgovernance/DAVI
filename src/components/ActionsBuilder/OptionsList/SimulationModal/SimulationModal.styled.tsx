import styled from 'styled-components';
import { Circle, Flex } from 'components/Primitives/Layout';

export const StatusWrapper = styled(Flex)`
  margin: 3rem 0px;
`;

export const SuccessCircle = styled(Circle)`
  color: ${({ theme }) => theme.colors.params['2']};
  border-color: ${({ theme }) => theme.colors.params['2']};
  background-color: transparent;
`;
export const WarningCircle = styled(Circle)`
  color: ${({ theme }) => theme.colors.red};
  border-color: ${({ theme }) => theme.colors.red};
  background-color: transparent;
`;

export const Message = styled.h3`
  font-family: Inter;
  font-style: normal;
  font-weight: 600;
  font-size: 16px;
  color: ${({ theme }) => theme.colors.text};
  margin: 0;
  padding: 1.5rem 0 1rem 0;
`;

export const Muted = styled.p`
  font-family: Inter;
  font-size: 12px;
  font-weight: 600;
  line-height: 16px;
  letter-spacing: 0em;
  text-align: left;
  color: ${({ theme }) => theme.colors.proposalText.grey};
`;

export const TenderlyLogo = styled.img`
  margin-left: 0.5rem;
  height: 20px;
`;
