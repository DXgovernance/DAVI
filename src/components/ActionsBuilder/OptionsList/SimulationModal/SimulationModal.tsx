import { Flex } from 'components/Primitives/Layout';
import { Modal } from 'components/Primitives/Modal';
import PendingCircle from 'old-components/Guilds/common/PendingCircle';
import { FiX, FiCheck } from 'react-icons/fi';
import {
  Message,
  Muted,
  StatusWrapper,
  SuccessCircle,
  TenderlyLogo,
  WarningCircle,
} from './SimulationModal.styled';
import { Button } from 'old-components/Guilds/common/Button';
import { useTranslation } from 'react-i18next';
import tenderlyLogo from 'assets/images/tenderly-dark-mode.svg';
import { SimulationState } from '../types';

interface SimulationModalProps {
  isOpen: boolean;
  onDismiss: () => void;
  status: SimulationState;
}

export const SimulationModal: React.FC<SimulationModalProps> = ({
  isOpen,
  onDismiss,
  status,
}) => {
  const { t } = useTranslation();

  return (
    <Modal
      isOpen={isOpen}
      onDismiss={onDismiss}
      header={t('simulations.verifyingActions')}
      maxWidth={390}
      zIndex={1000}
    >
      <Flex padding={`1.5rem`}>
        <StatusWrapper>
          <SimulationStatusWrapper status={status} />
          <Flex direction="row">
            <Muted>{t('poweredBy')}</Muted> <TenderlyLogo src={tenderlyLogo} />
          </Flex>
        </StatusWrapper>
        <Button
          disabled={status === SimulationState.pending}
          fullWidth
          onClick={onDismiss}
        >
          {t('dismiss')}
        </Button>
      </Flex>
    </Modal>
  );
};

const SimulationStatusWrapper = ({ status }) => {
  const { t } = useTranslation();

  switch (status) {
    case SimulationState.pending:
      return (
        <Flex>
          <PendingCircle height="86px" width="86px" color={'white'} />
          <Message>{t('simulations.verifyingActions')}</Message>
        </Flex>
      );
    case SimulationState.allPassed:
      return (
        <Flex>
          <SuccessCircle>
            <FiCheck size={40} />
          </SuccessCircle>
          <Message>{t('simulations.actionsVerified')}</Message>
        </Flex>
      );
    case SimulationState.someFailed:
      return (
        <Flex>
          <WarningCircle>
            <FiX size={40} />
          </WarningCircle>
          <Message>{t('simulations.actionsFailed')}</Message>
        </Flex>
      );
    default:
      return (
        <Flex>
          <WarningCircle>
            <FiX size={40} />
          </WarningCircle>
          <Message>{t('simulations.error')}</Message>
        </Flex>
      );
  }
};
