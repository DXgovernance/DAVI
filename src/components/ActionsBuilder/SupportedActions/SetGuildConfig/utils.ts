import { FIELDS } from './constants';
import { BigNumber } from 'ethers';
import { SetGuildConfigFields, ControlField } from './types';

export const bn = (n: string | number | BigNumber) => BigNumber.from(n);

type UpdatedValuesReturn = {
  [key in ControlField]: {
    newValue: BigNumber;
    currentValue: BigNumber;
  };
};
export const getUpdatedValues = (
  current: SetGuildConfigFields,
  modifyed: SetGuildConfigFields
): UpdatedValuesReturn | {} => {
  if (!current || !modifyed) return {};
  return Object.keys(current).reduce((acc, key) => {
    if (!FIELDS.map(f => f.name).includes(key)) return acc;
    const currentValue = bn(current[key] || 0);
    const newValue = bn(modifyed[key] || 0);

    if (!currentValue.eq(newValue)) {
      return {
        ...acc,
        [key]: { newValue, currentValue },
      };
    }
    return acc;
  }, {});
};

export const pickUpdatedOrDefaultValue = (
  current: BigNumber,
  modifyed: BigNumber
): BigNumber => {
  if (BigNumber.isBigNumber(current)) {
    if (BigNumber.isBigNumber(modifyed) && !bn(modifyed).eq(current)) {
      return bn(modifyed);
    }
    return bn(current);
  } else if (BigNumber.isBigNumber(modifyed)) {
    return bn(modifyed);
  }
  return bn(0);
};
