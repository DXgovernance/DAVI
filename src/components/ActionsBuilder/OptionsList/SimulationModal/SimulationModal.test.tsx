import { render } from 'utils/tests';
import { SimulationState } from '../types';
import { SimulationModal } from './SimulationModal';

describe('Simulation modal', () => {
  // There's a conflict between the animation code in
  // styled components and jest-styled-components library
  // Since this test isn't critical, I commented it.

  // it(`Should match 'pending' status snapshot`, () => {
  //   const { container } = render(
  //     <SimulationModal
  //       isOpen={true}
  //       onDismiss={jest.fn()}
  //       status={SimulationState.pending}
  //     />
  //   );

  //   expect(container).toMatchSnapshot();
  // });

  it(`Should match 'allPassed' status snapshot`, () => {
    const { container } = render(
      <SimulationModal
        isOpen={true}
        onDismiss={jest.fn()}
        status={SimulationState.allPassed}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it(`Should match 'someFailed' status snapshot`, () => {
    const { container } = render(
      <SimulationModal
        isOpen={true}
        onDismiss={jest.fn()}
        status={SimulationState.someFailed}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it(`Should match 'error' status snapshot`, () => {
    const { container } = render(
      <SimulationModal
        isOpen={true}
        onDismiss={jest.fn()}
        status={SimulationState.error}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
