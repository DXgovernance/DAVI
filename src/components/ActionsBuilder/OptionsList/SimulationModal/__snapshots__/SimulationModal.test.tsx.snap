// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simulation modal Should match 'allPassed' status snapshot 1`] = `
<div>
   
   
</div>
`;

exports[`Simulation modal Should match 'error' status snapshot 1`] = `
<div>
   
   
</div>
`;

exports[`Simulation modal Should match 'someFailed' status snapshot 1`] = `
<div>
   
   
</div>
`;
